[nws321] iP

nws321 opened the pull request:
### **Boombotroz**

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Boombotroz frees your mind of having to remember things you need to do. It's,

text-based
easy to learn
F̶A̶S̶T̶ _SUPER FAST_ to use
All you need to do is,

download it from [here](https://github.com/nus-cs2103-AY2425S1/ip/pull/107).
double-click it.
add your tasks.
let it manage your tasks for you 😉
And it is **FREE**!

Features:
- [x] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

`public class Boombotroz {`
`   public static void main(String[] args) {`
`           new Boombotroz("../data.txt").run();`
`   }`
`}`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

nws321 committed:
Rename, Greet, Exit

nws321 committed:
Echo

nws321 committed:
Add, List

nws321 committed:
Mark as Done

nws321 committed:
ToDos, Events, Deadlines

nws321 committed:
Automated Text UI Testing

nws321 committed:
Handle Errors

nws321 committed:
Delete

nws321 committed:
Abstraction

nws321 committed:
Save

nws321 committed:
Dates and Times

nws321 committed:
A-MoreOOP

nws321 committed:
Organise into Packages

nws321 committed:
Merge commit '81a9c537afe8492675eb253968939364d65db091'

nws321 committed:
Add Gradle Support

nws321 committed:
Add JUnit Tests

nws321 committed:
Create a JAR File

nws321 committed:
Add JavaDoc Comments

nws321 committed:
Follow the Coding Standard

nws321 committed:
Find Keyword

nws321 committed:
Merge branch 'branch-A-CodingStandard'

nws321 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/Boombotroz/Parser.java
#	src/main/java/Boombotroz/Task.java
#	src/main/java/Boombotroz/TaskList.java

nws321 committed:
Load text properly

nws321 committed:
Use CheckStyle

G-Y-Chan commented on src/main/java/boombotroz/Deadline.java:
```diff
@@ -0,0 +1,49 @@
+ * Deals with DEADLINE typed task.
+ */
+public class Deadline extends Task {
+    private String time;
```
I think this variable name seems vague. Perhaps you can consider something like deadline?

G-Y-Chan commented on src/main/java/boombotroz/ToDo.java:
```diff
@@ -0,0 +1,35 @@
+     * @inheritDoc
+     */
+    @Override
+    public void hasDate(Ui ui) throws BoomException {}
```
From the name of this method, it sounds as though it is supposed to return a boolean value. If you are checking whether a valid date exists, perhaps you can consider renaming it to checkForValidDate()?

G-Y-Chan commented on src/main/java/boombotroz/Ui.java:
```diff
@@ -0,0 +1,134 @@
+    /**
+     * Throws exception when input invalid.
+     */
+    public void invalidInput() throws BoomException {
```
Your method name sounds like a noun. According to Java Coding standards provided, names representing methods should be verbs. Perhaps you can try to come up with a verb that better describes the behaviour of this method?

G-Y-Chan commented on src/main/java/boombotroz/Ui.java:
```diff
@@ -0,0 +1,134 @@
+     * @param taskInput task details.
+     * @throws BoomException If no task details given.
+     */
+    public void emptyTask(String taskInput) throws BoomException {
```
Maybe you can come up with a verb for this method's name that describes the intended action of this method?

G-Y-Chan reviewed (changes requested):
Overall, your code looks well structured, with descriptive JavaDoc comments that explain the behaviour of your classes and methods.
However, i noticed that you have multiple methods that check the validity of a certain input, such as the wrongDeadline(LocalDate, LocalDate) method. The names of these methods appear to be a noun, which deviates from the Java Coding standards which recommends using verbs for methods. You may wish to consider modifying the names of these methods for better clarity of the behaviour of certain methods.
Otherwise, well done on your chatbot! 👍👍

nws321 committed:
Add a GUI to the chatbot

nws321 committed:
Organise UI

nws321 committed:
Add assertions

Introduce assertions across various parts of the code to ensure input
validity and to maintain internal state consistency.

Assertions help in identifying erroneous inputs and invalid
states early during the execution, preventing unexpected behaviour.

nws321 committed:
Improve code quality

Code had some areas that were lengthy and not as easily understandable.

Performing a critical review of the code and improving overall code
quality. For example, making happy path prominent, minimising code
duplication.

nws321 committed:
Merge pull request #1 from nws321/branch-A-Assertions

Add assertions

nws321 committed:
Merge branch 'master' into branch-A-CodeQuality

nws321 committed:
Merge pull request #2 from nws321/branch-A-CodeQuality

Improve code quality

nws321 committed:
Add priority level to tasks

Different tasks have different levels of importance that needs to be
indicated.

Additional check box to indicate level of priority on a scale of 1 to 5
added for user to see and know task priority.

nws321 committed:
Rectify bugs

Issues with date time was observed during testing.
Had task added to list when it should have thrown error

nws321 committed:
Fix priority error

nws321 committed:
Sort list by priority level

nws321 committed:
Beautify GUI

nws321 committed:
Include the Ui image

nws321 committed:
Add a User Guide

nws321 committed:
Follow Coding Standards

nws321 committed:
Improve README

nws321 committed:
Fix file creation
